### PR TITLE
Add draft version of the Gradle plug-in

### PR DESCRIPTION
**Pull request by vlad-lubenskyi**
This Gradle plug-in will help our customers add JxBrowser to the Gradle project. 

Usage example:

```
plugins {
    id("com.teamdev.jxbrowser.gradle") version "0.0.1"
}

dependencies {
    // Adds `jxbrowser-core`
    implementation(jxbrowser.core())

    // Adds `jxbrowser-swing`
    implementation(jxbrowser.swing())

    // Detects the current platform and adds necessary binaries
    implementation(jxbrowser.currentPlatform())

    // Adds only `jxbrowser-win64`
    implementation(jxbrowser.win64())
}

jxbrowser {
    version = "7.21.2"

    // It's either US or EUROPE.
    repositoryLocation = US

    // For clients that copy JxBrowser to internal repositories. This field overrides `repositoryLocation`.
    repositoryUrl = "http://custom/repository"

    // Call if you want EAP repository too.
   includePreviewBuilds()
}
```

### CODE REVIEW

File: src/main/java/com/teamdev/jxbrowser/gradle/JxBrowserDepsPlugin.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+        project.getRepositories().maven(repository -> {
+            repository.setUrl(project.getProviders().provider(() -> {
+                if (extension.includePreviewBuilds) {
+                    return EAP_REPOSITORY;
+                } else {
+                    return "";
+                }
+            }));
+        });
```
Comment (ivan-diachenko-td) [inline src/main/java/com/teamdev/jxbrowser/gradle/JxBrowserDepsPlugin.java:56]: Let's extract to a separate method and use ternary operator:
```java
repositoryUrl(extension.repositoryUrl != null ? extension.repositoryUrl : extension.repositoryLocation.url());
repositoryUrl(extension.includePreviewBuilds ? EAP_REPOSITORY : "");
```
Actually, is setting an empty URL ok?

File: src/main/resources/com.teamdev.jxbrowser.gradle.properties [hunk 2]
```diff
@@ -0,0 +1 @@
+implementation-class=com.teamdev.jxbrowser.gradle.JxBrowserDepsPlugin
```
Comment (ivan-diachenko-td) [inline src/main/resources/com.teamdev.jxbrowser.gradle.properties:1]: Empty line

File: src/main/java/com/teamdev/jxbrowser/gradle/Environment.java [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    }
+
+    /**
+     * Returns {@code true} if the current operating system is macOS.
+     */
+    public static boolean isMac() {
+        return osName().startsWith("Mac");
+    }
+
+
```
Comment (ivan-diachenko-td) [inline src/main/java/com/teamdev/jxbrowser/gradle/Environment.java:28]: Extra empty line

File: src/main/java/com/teamdev/jxbrowser/gradle/JxBrowserDeps.java [hunk 4]
```diff
@@ -0,0 +1,171 @@
+package com.teamdev.jxbrowser.gradle;
```
Comment (ivan-diachenko-td) [inline src/main/java/com/teamdev/jxbrowser/gradle/JxBrowserDeps.java:1]: Let' s add copyright

**Review by ivan-diachenko-td — CHANGES_REQUESTED** [review]

File: src/main/java/com/teamdev/jxbrowser/gradle/JxBrowserDepsPlugin.java (hunk 1 above)
Comment (vlad-lubenskyi) [reply src/main/java/com/teamdev/jxbrowser/gradle/JxBrowserDepsPlugin.java]: Not needed anymore.

**Pull request merged**